Add WebComponents flavoured VanillaJS implementation

localvoid opened the pull request:
This implementation should show performance overhead of WebComponents. It is implemented in imperative VanillaJS style.

Works in Chrome >= 67 ([Customized built-in elements](https://www.chromestatus.com/feature/4670146924773376))

krausest commented:
Thanks. But I'm afraid I've no results today due to #405
Currently only Chrome <= 66 works on my macbook, chrome 67 with chromedriver renders but shows no text.

krausest commented:
Thanks. Now that I found that I just have to set the LANG variable I was able to run a few frameworks  for chrome 67:
<img width="540" alt="bildschirmfoto 2018-06-13 um 20 38 01" src="https://user-images.githubusercontent.com/3374055/41371123-b23464e4-6f49-11e8-9610-c4aab67617f7.png">
@ryansolid @leeoniya @adamhaile 
Seems like chrome 67 is much faster for clear rows and using RAF isn't giving an advantage anymore (hurray for ending this discussion :sweat_smile:).
I'll try to re-run all frameworks the next days.

localvoid commented:
Thanks, this rAF hack is bizarre, I think that it is still can give an advantage, so it is better if everyone just removes it. Here is a run on my machine (Linux, Chrome 67) with some modifications to benchmark implementations.

![benchmark](https://user-images.githubusercontent.com/132045/41475250-75974498-70e8-11e8-934a-d9bd086e8013.png)

**surplus-EVENTS** doesn't have rAF hack and has a noticeable difference in clear rows.

Modifications:

- [ivi-SIMPLE](https://github.com/localvoid/js-framework-benchmark/blob/sandbox/frameworks/ivi-SIMPLE/src/main.js) - Event delegation, removed connectors, removed stateful components
- [surplus-EVENTS](https://github.com/localvoid/js-framework-benchmark/blob/sandbox/frameworks/surplus-EVENTS/src/view.tsx) - Removed event delegation, removed rAF hack
- [Inferno-SC](https://github.com/localvoid/js-framework-benchmark/blob/sandbox/frameworks/inferno-SC/src/controller.jsx) - Each row is a stateful component

https://rawgit.com/localvoid/js-framework-benchmark/sandbox/webdriver-ts-results/table.html

ryansolid commented:
Yeah It's better but it's still there. I think it's just that the more powerful the computer the smaller the difference gets. I am seeing about a 10ms on my machine. It was like 35ms before though.. I think on the machine that run the benchmark it accounts for about a 5ms difference.  Getting to the point it doesn't particularly matter though. But as indicated above all those libraries are so close.
